### ISSUE TITLE
Not a cross-platform solution

### ISSUE THREAD

**Issue by gazpachoking** [gazpachoking/pdm-buildlocked#3]
This might not be so clever of a solution as I thought. There are probably failure cases when dependencies have environment markers where the pinned deps end up wrong. I don't know the exact cases, and it seems to work for dependency trees that don't get too complicated. See https://github.com/pdm-project/pdm/issues/1476

**Comment by lecardozo** [comment]
@gazpachoking that shouldn't be a problem if we are building the distribution in the same platform in which it's supposed to be used, right?

**Comment by gazpachoking** [comment]
> @gazpachoking that shouldn't be a problem if we are building the distribution in the same platform in which it's supposed to be used, right?

I don't believe so. I haven't figured out the full implications of the shortcoming, but I think in most cases it should still work fine. (And even in many cross-platform situations it should still work.)
